jujugui/static/gui/src/app/store/env/controller-api.js

```diff
@@ -387,6 +393,28 @@ YUI.add('juju-controller-api', function(Y) {
     },
 
     /**
+      Retrieve bundle changes from the bundle service.
+
+      @method _getBundleChangesFromBundleService
+      @param {String} bundleYAML The bundle YAML file contents.
+      @param {Function} callback The user supplied callback to send the bundle
+        changes response to after proper post processing. The callback receives
+        a list of errors (each one being a string describing a possible error)
+        and a list of bundle changes.
+    */
+    _getBundleChangesFromBundleService: function(bundleYAML, callback) {
+      this.get('bundleService').getBundleChangesFromYAML(
+        bundleYAML, (error, changes) => {
+          if (error) {
+            callback([error], []);
+            return;
+          }
+          callback([], changes);
```

Comment:
If there are no errors then the error position in the callback should be `null`. I know that this callback expects an array but that fix should be propagated up and down.